### PR TITLE
Add type hints to Redis

### PR DESCRIPTION
**Pull request by Kludex**

### CODE REVIEW

**Comment by xrmx** [comment]
We have a pylint failure here

**Comment by Kludex** [comment]
> We have a pylint failure here

fixed 👀 👍

**Review by emdneto — APPROVED** [review]

File: instrumentation/opentelemetry-instrumentation-redis/src/opentelemetry/instrumentation/redis/__init__.py [hunk 1]
```diff
@@ -109,18 +110,43 @@ def response_hook(span, instance, response):
...
 from opentelemetry.instrumentation.utils import unwrap
 from opentelemetry.semconv.trace import SpanAttributes
-from opentelemetry.trace import Span, StatusCode
+from opentelemetry.trace import Span, StatusCode, Tracer
 
-_DEFAULT_SERVICE = "redis"
+if TYPE_CHECKING:
+    from typing import Awaitable, TypeVar
 
-_RequestHookT = typing.Optional[
```
Comment (lzchen) [inline instrumentation/opentelemetry-instrumentation-redis/src/opentelemetry/instrumentation/redis/__init__.py:116]: Any reason for removing the `Optional` typing?

**Review by lzchen — APPROVED** [review]
Big thanks for this!

**Pull request merged**